### PR TITLE
fix complie error with vs2013

### PR DESCRIPTION
**Pull request by yajiedesign**
fix complie error with vs2013
fix lint with windows

### CODE REVIEW

File: scripts/lint.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def get_header_guard_dmlc(filename):
         file_path_from_root = _HELPER.project_name +  file_path_from_root[idx + 3:]
     else:
         for spath in inc_list:
-            prefix = spath + os.sep
+            prefix = spath + '/'
```
Comment (piiswrong) [inline scripts/lint.py]: why?

Comment (yajiedesign) [inline scripts/lint.py]: because  in windows,this path also use '/'

File: include/dmlc/optional.h [hunk 2]
```diff
@@ -15,10 +15,16 @@
 #include "./type_traits.h"
 
 namespace dmlc {
+
 /*! \brief dummy type for assign null to optional */
 struct nullopt_t {
+#if defined(_MSC_VER) && _MSC_VER < 1900
+  /*! \brief dummy constructor */
+  explicit nullopt_t(int) {}
+#elif
```
Comment (piiswrong) [inline include/dmlc/optional.h]: did you mean else?

Comment (yajiedesign) [inline include/dmlc/optional.h]: yes....strange, I remember I writing else

**Pull request merged**